packages/slider/slider-thumb.ts

```diff
@@ -0,0 +1,323 @@
+   * @internal
+   */
+  anchorPositioningStyleElement: HTMLStyleElement | null = null;
+
+  #showTooltip(): void {
+    this._showTooltip = true;
+  }
+
+  #hideTooltip(): void {
+    this._showTooltip = false;
+  }
+
+  #onInput(e: InputEvent | CustomEvent): boolean {
+    const textInputValue = (e as CustomEvent<{ value: string }>).detail?.value;
+    if (textInputValue && this._invalid) this._invalid = false;
+
+    const value = textInputValue || (e.currentTarget as WarpSliderThumb).value;
+    const valueNum = Number.parseInt(value);
+
+    // Update validation state
+    // Check that the user hasn't typed in a value beyond max or min
+    const maxNum = Number.parseInt(this.max);
+    const minNum = Number.parseInt(this.min);
+    if (valueNum > maxNum || valueNum < minNum) {
+      this._invalid = true;
+    }
+
+    // Stop a range slider's from value from reaching past the to value and vice versa
+    // by updating the other component's min and max values.
+    let shouldCancel = false;
+    if (this.slot) {
+      const computedStyle = getComputedStyle(this);
+      if (this.slot === 'from') {
+        const toValue = computedStyle.getPropertyValue('--to');
+        // Check that the from value is not about to go past the --to value
+        if (valueNum > Number.parseInt(toValue)) {
+          if (textInputValue) {
+            this._invalid = true;
+            return false;
+          }
+          shouldCancel = true;
+          // The user might have moved the slider so fast that this.value is far away from overlapping.
+          // Set it to be equal to the to/from value, depending on what slider the user's moving.
+          this.value = toValue;
+        }
+      } else {
+        const fromValue = computedStyle.getPropertyValue('--from');
+        // Check that the to value is not about to go past the --from value
+        if (valueNum < Number.parseInt(fromValue)) {
+          if (textInputValue) {
+            this._invalid = true;
+            return false;
+          }
+          shouldCancel = true;
+          // The user might have moved the slider so fast that this.value is far away from overlapping.
+          // Set it to be equal to the to/from value, depending on what slider the user's moving.
+          this.value = fromValue;
+        }
+      }
+    }
+
+    if (shouldCancel) {
+      e.preventDefault();
+      // Needed to stop slider from moving independendtly of the value when we cancel the event
+      this.range.value = this.value;
+      return false;
+    }
+    this.value = value;
+    return true;
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+    if (!('anchorName' in document.documentElement.style)) {
+      // Load the polyfill for CSS anchor positioning by @oddbird for browsers without native support.
+      const dirname = import.meta.url.substring(0, import.meta.url.lastIndexOf('/'));
+
+      const [{ default: polyfill }] = await Promise.all([
```

Comment:
Should you handle promise rejection here? Else it'll be silent and impossible to debug, if it ever fails :)